hw4/hw4.cpp

```diff
@@ -9,32 +9,48 @@
 __global__ void histogram_smem_atomics(const float *in, int width, int height, float *out)
 {
     // pixel coordinates
-    int binSize = 2;
-    int x = blockIdx.x * blockDim.x + threadIdx.x;
-    int y = blockIdx.y * blockDim.y + threadIdx.y;
-    
-    // linear thread index within 2D block
-    int t = threadIdx.x + threadIdx.y * blockDim.x; 
-    //int t = threadIdx.x + blockDim.x*blockIdx.x;
+    int binSize = width/NUM_BINS;
     
+    __shared__ unsigned int numThreads;
     // initialize temporary accumulation array in shared memory
     __shared__ unsigned int smem[NUM_BINS];
-    __shared__ unsigned int numThreads;
+    numThreads = 0;
+    //int x = blockIdx.x * blockDim.x + threadIdx.x;
+    //int y = blockIdx.y * blockDim.y + threadIdx.y;
+    
+    // linear thread index within 2D block
+    int t = threadIdx.x + threadIdx.y * blockDim.x;
+    int blockId = blockIdx.y * gridDim.x + blockIdx.x;
+    int threadId = blockId * blockDim.x + threadIdx.x;
+    t = threadIdx.x + blockIdx.x * blockDim.x;
+
+    if(blockIdx.x ==2)
+    {
+       //atomicAdd(&out[1023],1);
+    }
+   
+   
+    
+    
     if(t<NUM_BINS)
     {
        smem[t] = 0;
     }
-    numThreads = 0;
+     
+    
+    
+    
+    
     
     __syncthreads();
+
     atomicAdd(&numThreads,1);
+    //atomicAdd(&out[t],1);
     
-    // process pixels
-    // updates our block's partial histogram in shared memory
-    unsigned int rgb;
-    //rgb = (unsigned int)(in[y * width + x]); //Numbers between 0 and 1024
-    rgb = (unsigned int)(in[t]);
-    atomicAdd(&smem[rgb/binSize], 1);
+    unsigned int temp;
+    temp = (unsigned int)(in[t]);
+    //atomicAdd(&smem[temp/binSize], 1);
+    atomicAdd(&out[temp/binSize],1);
```

Comment:
I think you should look at the histo.cpp demo instead of the color_histo for this assignment. The data is 1D so there's a lot of things that we don't need to consider. https://github.com/stevenireeves/GPU_Computing/blob/master/hip/color_histo/histo.cpp#L33-L54